src/node/rpc/member_frontend.h

```diff
@@ -387,22 +394,65 @@ namespace ccf
         caller_id, {cose_sign1.begin(), cose_sign1.end()});
     }
 
-    bool is_proposal_replayed_or_stale(
+    ProposalSubmissionStatus is_proposal_submission_acceptable(
       kv::Tx& tx,
-      uint64_t created_at,
+      const std::string& created_at,
       const std::vector<uint8_t>& request_digest,
-      const std::string& proposal_id)
+      const ccf::ProposalId& proposal_id,
+      ccf::ProposalId& colliding_proposal_id,
+      std::string& min_created_at)
     {
       auto cose_recent_proposals = tx.rw(network.cose_recent_proposals);
       auto key = fmt::format("{}:{}", created_at, ds::to_hex(request_digest));
+
+      std::vector<std::string> replay_keys;
+      cose_recent_proposals->foreach_key(
+        [&replay_keys](const std::string& replay_key) {
+          replay_keys.push_back(replay_key);
+          return true;
+        });
+
+      std::sort(replay_keys.begin(), replay_keys.end());
+
+      // New proposal must be more recent than median proposal kept
+      if (!replay_keys.empty())
+      {
+        auto [min_created_at, _] =
+          nonstd::split_1(replay_keys[replay_keys.size() / 2], ":");
```

Comment:
This `min_created_at` is shadowing the out-parameter, and looks like it unused at the call-site. It should be formatted into the `TooOld` error message.

Reply:
I meant to do that (hence the out parameter), but apparently forgot. Thank you!